File: Automation_App_Frontend/lib/features/email_versand/domain/entities/outlook_signatur.dart
import 'package:equatable/equatable.dart';

/// Eine im Mailprogramm eingerichtete Signatur (§4.7). Der [name] ist der, unter
/// dem sie in Outlook steht — eine Kanzlei hat oft mehrere („Kurz",
/// „Vollständig"), und welche gemeint ist, kann nur der Anwalt sagen.
class OutlookSignatur extends Equatable {
  final String name;
  final String text;

  /// True, wenn daneben eine formatierte Fassung liegt — Schrift, Farben,
  /// Logo. Die App übernimmt sie mit; wie schwer ihre Bilder wiegen, zeigt
  /// sich erst danach.
  final bool hatFormat;

  const OutlookSignatur({
    required this.name,
    required this.text,
    this.hatFormat = false,
  });

  factory OutlookSignatur.fromJson(Map<String, dynamic> json) {
    return OutlookSignatur(
      name: json['name'] as String? ?? '',
      text: json['text'] as String? ?? '',
      hatFormat: json['hatFormat'] as bool? ?? false,
    );
  }

  /// Die ersten Zeilen, um sie in der Auswahl auseinanderzuhalten.
  String get vorschau {
    final zeilen = text
        .split('\n')
        .map((zeile) => zeile.trim())
        .where((zeile) => zeile.isNotEmpty)
        .take(2);
    return zeilen.join(' · ');
  }

  @override
  List<Object?> get props => [name, text, hatFormat];
}

File: AutomationService/AutomationService/Features/EmailVersand/Domain/Services/OutlookSignatur.cs
namespace AutomationService.Features.EmailVersand.Domain.Services;

/// <summary>
/// Eine im Mailprogramm eingerichtete Signatur, zur Übernahme in die
/// Einstellungen (§4.7). Der Name ist der, unter dem sie in Outlook steht —
/// eine Kanzlei hat oft mehrere („Kurz", „Vollständig"), und welche gemeint
/// ist, kann nur der Anwalt sagen.
/// </summary>
/// <param name="Name">Anzeigename in Outlook.</param>
/// <param name="Text">Outlooks eigene Nur-Text-Fassung, für die Vorschau.</param>
/// <param name="HatFormat">
/// True, wenn daneben eine formatierte Fassung liegt (Schrift, Farben, Logo).
/// Wie schwer deren Bilder wiegen, sagt erst die Übernahme: Dafür müssten sie
/// hier alle gelesen werden, nur um eine Liste anzuzeigen.
/// </param>
public sealed record OutlookSignatur(string Name, string Text, bool HatFormat = false);
